src/openms/source/ANALYSIS/ID/IDScoreGetterSetter.cpp

```diff
@@ -68,7 +68,60 @@ namespace OpenMS
     }
   }
 
+  inline bool isFirstBetterScore_(double first, double second, bool isHigherBetter)
+  {
+    if (isHigherBetter) return first > second; else return first < second;
+  }
+
+  inline void addToPeptideScoreMap_(
+    std::unordered_map<String, ScoreToTgtDecLabelPair>& seq_to_score_labels,
+    const PeptideIdentification& id)
+  {
+    bool higher_better = id.isHigherScoreBetter();
+    if (id.getHits().empty())
+    {
+      return;
+    }
+    const auto& best_hit = id.getHits()[0];
+    double score = best_hit.getScore();
+    auto [it, found] = seq_to_score_labels.try_emplace(
+      best_hit.getSequence().toUnmodifiedString(),
+      score,
+      (best_hit.getMetaValue("target_decoy") != DataValue::EMPTY) &&
+        (best_hit.getMetaValue("target_decoy").toString().hasPrefix("target")));
 
+    if (found && isFirstBetterScore_(score, it->second.first, higher_better))
+    {
+      it->second.first = score;
+    }
+  }
+
+  void IDScoreGetterSetter::fillPeptideScoreMap_(
+    std::unordered_map<String, ScoreToTgtDecLabelPair>& seq_to_score_labels,
+    const vector<PeptideIdentification>& ids)
+  {
+    for (auto const & id : ids)
+    {
+      addToPeptideScoreMap_(seq_to_score_labels, id);
+    }
+  }
+
+  void IDScoreGetterSetter::fillPeptideScoreMap_(
+    std::unordered_map<String, ScoreToTgtDecLabelPair>& seq_to_score_labels,
+    ConsensusMap const& map,
+    bool include_unassigned = true)
+  {
+    map.applyFunctionOnPeptideIDs(
+      [&seq_to_score_labels](const PeptideIdentification& id){addToPeptideScoreMap_(seq_to_score_labels, id);},
+      include_unassigned);
+  }
+
+  /**
+   * @ingroup getScoresFunctions
+   * @brief For protein groups. Groups are target if at least one protein is target
+   * Decoy accessions are determined by the decoy substring.
+   * Uses the "picked" algorithm. As soon as there was one target picked over a
```

Comment:
incomplete sentence at end

Reply:
```suggestion
   * Uses the "picked" algorithm. As soon as there was one member which was picked as target over a decoy, the group is counted as target. Otherwise as decoy.
```